.github/workflows/build-windows-package.yml

```diff
@@ -110,31 +112,22 @@ jobs:
         with:
           filenames: php_mongodb.dll
 
-      # Copy the signature file from the release asset directory to avoid directory issues in the ZIP file
-      - name: "Copy signature file"
-        run: cp ${RELEASE_ASSETS}/php_mongodb.dll.sig .
+      - name: "Generate file name for DLL and archive"
+        run:
+          echo FILENAME="php_mongodb-${{ inputs.version }}-${{ inputs.php }}-${{ inputs.ts }}-${{ needs.build.outputs.vs }}-${{ inputs.arch == 'x64' && 'x64_86' || inputs.arch }}" >> "$GITHUB_ENV"
 
-      - name: "Set compiler environment variable"
+      # In this step, we:
+      # - update the extension DLL file name to match the expectation of pie
+      # - copy the signature file from the release asset directory to avoid directory issues in the archive
+      # - rename the signature file to match the extension DLL file
+      - name: "Copy signature file and use correct file names"
         run: |
-          case "$PHP_VERSION" in
-            "7.4")
-              COMPILER="vc15"
-              ;;
-            "8.0" | "8.1" | "8.2" | "8.3")
-              COMPILER="vs16"
-              ;;
-            "8.4")
-              COMPILER="vs17"
-              ;;
-          esac
-          echo "COMPILER=${COMPILER}" >> "$GITHUB_ENV"
-        shell: bash
-        env:
-          PHP_VERSION: ${{ inputs.php }}
+          mv php_mongodb.dll ${{ env.FILENAME }}.dll
+          cp ${RELEASE_ASSETS}/php_mongodb.dll.sig ${{ env.FILENAME }}.dll.sig
 
       - name: "Create and upload release asset"
         if: ${{ inputs.upload_release_asset }}
         run: |
-          ARCHIVE=php_mongodb-${{ inputs.version }}-${{ inputs.php }}-${{ inputs.ts }}-${{ env.COMPILER }}-${{ inputs.arch }}.zip
-          zip ${ARCHIVE} php_mongodb.dll php_mongodb.dll.sig php_mongodb.pdb CREDITS CONTRIBUTING.md LICENSE README.md THIRD_PARTY_NOTICES
+          ARCHIVE=${{ env.FILENAME }}.zip
+          zip ${ARCHIVE} ${{ env.FILENAME }}.dll ${{ env.FILENAME }}.dll.sig php_mongodb.pdb CREDITS CONTRIBUTING.md LICENSE README.md THIRD_PARTY_NOTICES
```

Comment:
I'm not sure, but I think `php_mongodb.pdb` needs to become `${{ env.FILENAME }}.pdb`.

See: https://github.com/php/pie/blob/eaf77ea181f44c7711887474c4f33501a17cb39f/src/Installing/WindowsInstall.php#L39

Reply:
Hmm, according to the [maintainer docs](https://github.com/php/pie/blob/main/docs/extension-maintainers.md#contents-of-the-windows-zip), it doesn't:

> `php_{extension-name}.pdb` - this will be moved alongside the `C:\path\to\php\ext\php_{extension-name}.dll`

Looking at the code you linked, I do agree that it should be named the same way as the DLL file.

Let me try summoning @asgrim to clarify 😉 

---

Another concern that comes to mind is that we're now requiring people that install their extensions manually. Right now, they only need to copy `php_mongodb.dll` to their extensions directory and they're done. This is valid regardless of whether they download our packages or the ones offered on pecl.php.net.

I'm sure there was a particular reason for naming the DLL file this way, but I wonder if that's truly necessary. Clearly, we can extract the relevant information (compiler, thread safety, PHP version, architecture) from the archive filename before we extract the DLL file, so we shouldn't need to re-verify this when we have the DLL file. If a user packs a DLL file into an archive for a different platform, that's their problem, not PIE's.

Reply:
Regarding the first part, yes the implementation is incorrect, it was always meant to be `php_{extension-name}.pdb` - I've reported issue php/pie#71 and I will look into getting this fixed.

The second part; the naming was agreed since it aligns with what [php/php-windows-builder](https://github.com/php/php-windows-builder) action produces